javascript/load-gen.ts

```diff
@@ -0,0 +1,406 @@
+  ): string {
+    return (
+      Math.round((count / context.globalRequestCount) * 100 * 10) / 10
+    ).toString();
+  }
+
+  private static outputHistogramSummary(histogram: hdr.Histogram): string {
+    return `
+  count: ${histogram.totalCount} 
+    min: ${histogram.minNonZeroValue}
+    p50: ${histogram.getValueAtPercentile(50)}
+    p90: ${histogram.getValueAtPercentile(90)}
+    p99: ${histogram.getValueAtPercentile(99)}
+    max: ${histogram.maxValue}
+`;
+  }
+
+  private static getElapsedMillis(startTime: [number, number]): number {
+    const endTime = process.hrtime(startTime);
+    return (endTime[0] * 1e9 + endTime[1]) / 1e6;
+  }
+}
+
+const PERFORMANCE_INFORMATION_MESSAGE = `
+Thanks for trying out our basic javascript load generator!  This tool is
+included to allow you to experiment with performance in your environment
+based on different configurations.  It's very simplistic, and only intended
+to give you a quick way to explore the performance of the Momento client
+running on a single nodejs process.
+
+Note that because nodejs javascript code runs on a single thread, the limiting
+factor in request throughput will often be CPU.  Keep an eye on your CPU
+consumption while running the load generator, and if you reach 100%
+of a CPU core then you most likely won't be able to improve performance further
+without running additional nodejs processes.
+
+Also, since performance will be impacted by latency, you'll get the best results
+if you run on a cloud VM in the same region as your Momento cache.
+
+Check out the configuration settings at the bottom of the 'load-gen.ts' to
+see how different configurations impact performance.
+
+If you have questions or need help experimenting further, please reach out to us!
+`;
+
+async function main(loadGeneratorOptions: BasicJavaScriptLoadGenOptions) {
+  const loadGenerator = new BasicJavaScriptLoadGen(loadGeneratorOptions);
+  await loadGenerator.run();
+}
+
+const loadGeneratorOptions: BasicJavaScriptLoadGenOptions = {
+  /**
+   * This setting allows you to control the verbosity of the log output during
+   * the load generator run.
+   */
+  loggerOptions: {
+    /**
+     * Available log levels are TRACE, DEBUG, INFO, WARN, and ERROR.  DEBUG
+     * is a reasonable choice for this load generator program.
+     */
+    level: LogLevel.DEBUG,
+    /**
+     * Allows you to choose between formatting your log output as JSON (a good
+     * choice for production environments) or CONSOLE (a better choice for
+     * development environments).
+     */
+    format: LogFormat.CONSOLE,
+  },
+  /**
+   * Configures the Momento client to timeout if a request exceeds this limit.
+   * Momento client default is 5 seconds.
+   */
+  requestTimeoutMs: 5 * 1000,
+  /**
+   * Controls the size of the payload that will be used for the cache items in
+   * the load test.  Smaller payloads will generally provide lower latencies than
+   * larger payloads.
+   */
+  cacheItemPayloadBytes: 2500,
```

Comment:
NIT: can we default to smaller payload (better latencies), 100 bytes is my suggestion

Reply:
done in d423998132f04e68fb33060da3821826b95fc37c .  On my machine with the new default settings you can get about 1300 tps with p50 of 17ms.